core-bundle/tests/Cache/EntityCacheTagsTest.php

```diff
@@ -258,10 +258,9 @@ public function testDelegatesToCacheInvalidator(): void
 
     private function getEntityCacheTags(ResponseTagger $responseTagger = null, CacheInvalidator $cacheInvalidator = null): EntityCacheTags
     {
-        return new EntityCacheTags(
-            $this->getTestEntityManager(),
-            $responseTagger ?? $this->createMock(ResponseTagger::class),
-            $cacheInvalidator ?? $this->createMock(CacheInvalidator::class)
-        );
+        $responseTagger ??= $this->createMock(ResponseTagger::class);
+        $cacheInvalidator ??= $this->createMock(CacheInvalidator::class);
+
+        return new EntityCacheTags($this->getTestEntityManager(), $responseTagger, $cacheInvalidator);
```

Comment:
IMHO the old version was easier to read. :see_no_evil: 


I think I prefer this…

```php 
doFoo(
    $firstArgument ?? new Fallback(),
    $secondArgument ?? $alternative,
);
```

over this

```php 
$firstArgument ??= new Fallback();
$secondArgument ??= $alternative;

doFoo($firstArgument, $secondArgument);
```

in general.

It's basically the same amount of code but with the first option, the place where the "fallback chain" is put, is exactly the place it is used. The code basically reads "use this as the argument, fallback to this, fallback to that, …".

In the second one, you do not know the usage (yet) when reading the assignments. Another point: Why would you overwrite an argument if not needed? WIth `$a ??= $b` the original value of `$a` is gone.

Reply:
I think it could also be read as default values for the parameters, which is where the `$this->createMock(…)` would go if PHP would support expressions as default values. From that perspective using `??=` at the beginning of the method makes sense IMO.

But I would intuitively also write it in the style the old version was.
But I’m happy either way ☺️

Reply:
> Another point: Why would you overwrite an argument if not needed? WIth `$a ??= $b` the original value of `$a` is gone.

That's not a point, because the same happens with the old `if (null === $a) { $a = $b; }` code.

Reply:
I have reverted some of the changes in 99c07707065b4e90a76685af287ca03db8156e04 to explicitly allow the `$var ?? $this->createMock()` construct.